modules/tinymce/src/plugins/link/test/ts/browser/NormalizeLinkTest.ts

```diff
@@ -0,0 +1,120 @@
+    });
+
+    it('TINY-6508: Multiple short ended elements', () => {
+      test('<p><a href="tiny">test</a><br><img><br></p>', makeRange([ 0, 0, 0 ], 0, [ 0 ], 3), Optional.none());
+    });
+
+    it('TINY-6508: Nest link', () => {
+      test('<p><b><a href="tiny">test</a></b><br></p>', makeRange([ 0, 0, 0, 0 ], 0, [ 0 ], 1), Optional.some([ 0, 0, 0 ]));
+    });
+
+    it('TINY-6508: Nest link next to another Nest link', () => {
+      test('<p><b><a href="tiny">test</a></b><b><a href="tiny">tiny</a></b><br></p>', makeRange([ 0, 0, 0, 0 ], 0, [ 0 ], 2), Optional.none());
+    });
+
+    it('TINY-6508: Single link with one level nest child', () => {
+      test('<p><a href="tiny"><strong>word</strong>other</a><br></p>', makeRange([ 0, 0, 0, 0 ], 0, [ 0 ], 1), Optional.some([ 0, 0 ]));
+    });
+
+    it('TINY-6508: Single link with two level nest child', () => {
+      test('<p><a href="tiny"><strong><em>word<em/></strong>other</a><br></p>', makeRange([ 0, 0, 0, 0, 0 ], 0, [ 0 ], 1), Optional.some([ 0, 0 ]));
+    });
+
+    it('TINY-6508: Nest link at the end of the line', () => {
+      test('<p><em>test</em><b><a href="tiny">test</a></b><br></p>', makeRange([ 0, 1, 0 ], 0, [ 0 ], 2), Optional.some([ 0, 1, 0 ]));
+    });
+
+    it('TINY-6508: Multiple links', () => {
+      test('<p><a href="tiny">link1</a>content<a href="tiny">link2</a><br></p>', makeRange([ 0, 0, 0 ], 0, [ 0 ], 3), Optional.none());
+    });
+
+    it('TINY-6508: Simple link with nest Img', () => {
+      test('<p><a href="tiny"><img src="image.jpg"></a>text<br></p>', makeRange([ 0, 0, 0 ], 0, [ 0 ], 2), Optional.none());
+    });
+
+    it('TINY-6508: Selection end between two BR', () => {
+      test('<p>text<a href="tiny">test</a><br><br></p>', makeRange([ 0, 1, 0 ], 0, [ 0 ], 3), Optional.none());
+    });
+
+    it('TINY-6508: Range end in another block element (paragraph)', () => {
+      test('<p>text<a href="tiny">test</a><br></p><p>info<br></p>', makeRange([ 0, 1, 0 ], 0, [ 1 ], 1), Optional.none());
+    });
+
+    it('TINY-6508: Nest link in a block element (header)', () => {
+      test('<h1><a href="tiny">tiny</a><em>tiny</em><br></h1>', makeRange([ 0, 0, 0 ], 0, [ 0 ], 2), Optional.none());
+    });
+
+    it('TINY-6508: Range end in another block element (header)', () => {
+      test('<h1><a href="tiny">test</a><br></h1><h1>info<br></h1>', makeRange([ 0, 0, 0 ], 0, [ 1 ], 1), Optional.none());
+    });
+
+    it('TINY-6508: Nest link in a block element (div)', () => {
+      test('<div><a href="tiny">tiny</a><em>tiny</em><br></div>', makeRange([ 0, 0, 0 ], 0, [ 0 ], 2), Optional.none());
+    });
+
+    it('TINY-6508: Range end in another block element (div)', () => {
+      test('<div><a href="tiny">test</a><br></div><div>info<br></div>', makeRange([ 0, 0, 0 ], 0, [ 1 ], 1), Optional.none());
+    });
+  });
+
+  it('TINY-6508: Return a link if the selection starts in the link and end in the link text', () => {
```

Comment:
Maybe add that this is a IE case here as well. Maybe move it to it's own describe not sure.

Reply:
Add in a new describe for IE cases